### ISSUE TITLE
Bad optimization plan on simple query

### ISSUE THREAD

**Issue by DeShadow** [arangodb/arangodb#2450]
ArangoDB 3.2 alpha 3

Simple query:
```
let originalNews = DOCUMENT(news, @newsId)
let linkedNewsIds = originalNews.linkedNewsIds

for n in news
    filter n._key in linkedNewsIds
    return n
```

I think it's predicatable that ArangoDB will use primary index on this query,  but optimization plan shows full scan:
```
Query string:
 let originalNews = DOCUMENT(news, @newsId)
 let linkedNewsIds = originalNews.linkedNewsIds
 for n in news
     filter n._key in linkedNewsIds
     return n

Execution plan:
 Id   NodeType                    Est.   Comment
  1   SingletonNode                  1   * ROOT
  2   CalculationNode                1     - LET originalNews = DOCUMENT(news   /* all collection documents */, "123")   /* simple expression */
  4   EnumerateCollectionNode   176000     - FOR n IN news   /* full collection scan */
  5   CalculationNode           176000       - LET #3 = (n.`_key` in originalNews.`linkedNewsIds`)   /* simple expression */   /* collections used: n : news */
  6   FilterNode                176000       - FILTER #3
  7   ReturnNode                176000       - RETURN n

Indexes used:
 none

Optimization rules applied:
 Id   RuleName
  1   remove-unnecessary-calculations
```

Simple hack `NOOPT` or `V8` helps ArangoDB to use primary index:
```
Query string:
 let originalNews = DOCUMENT(news, @newsId)
 let linkedNewsIds = NOOPT(originalNews.linkedNewsIds)
 for n in news
     filter n._key in linkedNewsIds
     return n

Execution plan:
 Id   NodeType          Est.   Comment
  1   SingletonNode        1   * ROOT
  2   CalculationNode      1     - LET originalNews = DOCUMENT(news   /* all collection documents */, "123")   /* simple expression */
  3   CalculationNode      1     - LET linkedNewsIds = NOOPT(originalNews.`linkedNewsIds`)   /* simple expression */
  8   IndexNode            1     - FOR n IN news   /* primary index scan */
  7   ReturnNode           1       - RETURN n

Indexes used:
 By   Type      Collection   Unique   Sparse   Selectivity   Fields       Ranges
  8   primary   news         true     false       100.00 %   [ `_key` ]   (n.`_key` in linkedNewsIds)

Optimization rules applied:
 Id   RuleName
  1   use-indexes
  2   remove-filter-covered-by-index
  3   remove-unnecessary-calculations-2
```

**Comment by jsteemann** [comment]
Fixed in devel (target release: 3.2.alpha4)

**Comment by dothebart** [comment]
3.2 alpha 4 containing this bugfix is available for download now.
